flake.nix

```
@@ -4,51 +4,94 @@
+        inputs = { nix-darwin.follows = "darwin"; nixos-20_09.follows = "nixos"; nixos-unstable.follows = "latest"; };
       };
       darwin.url = "github:LnL7/nix-darwin";
-      darwin.inputs.nixpkgs.follows = "override";
-      deploy = {
-        url = "github:serokell/deploy-rs";
-        inputs = { flake-compat.follows = "flake-compat"; naersk.follows = "naersk"; nixpkgs.follows = "override"; utils.follows = "utils"; };
-      };
-      devshell.url = "github:numtide/devshell";
-      flake-compat.url = "github:BBBSnowball/flake-compat/pr-1";
-      flake-compat.flake = false;
+      darwin.inputs.nixpkgs.follows = "latest";
       home.url = "github:nix-community/home-manager";
       home.inputs.nixpkgs.follows = "nixos";
       naersk.url = "github:nmattia/naersk";
-      naersk.inputs.nixpkgs.follows = "override";
+      naersk.inputs.nixpkgs.follows = "latest";
       nixos-hardware.url = "github:nixos/nixos-hardware";
-      utils.url = "github:numtide/flake-utils";
+
       pkgs.url = "path:./pkgs";
       pkgs.inputs.nixpkgs.follows = "nixos";
     };
 
-  outputs = inputs@{ deploy, nixos, nur, self, utils, ... }:
-    let
-      lib = import ./lib { inherit self nixos utils inputs; };
-    in
-    lib.mkFlake
-      {
-        inherit self;
-        hosts = ./hosts;
-        packages = import ./pkgs;
-        suites = import ./profiles/suites.nix;
-        extern = import ./extern;
-        overrides = import ./extern/overrides.nix;
-        overlays = ./overlays;
-        profiles = ./profiles;
-        userProfiles = ./users/profiles;
-        modules = import ./modules/module-list.nix;
-        userModules = import ./users/modules/module-list.nix;
-      } // {
-      inherit lib;
-      defaultTemplate = self.templates.flk;
+  outputs = inputs@{ self, pkgs, devlib, nixos, ci-agent, home, nixos-hardware, nur, ... }:
+    devlib.lib.mkFlake {
+      inherit self inputs;
+
+      channelsConfig = { allowUnfree = true; };
+
+      channels = {
+        nixos = {
+          overlays =
+            (devlib.lib.importers.pathsIn ./overlays) ++
+            [
+              ./pkgs/default.nix
+              pkgs.overlay # for `srcs`
+              nur.overlay
+            ];
+        };
+        latest = { };
+      };
+
+      lib = import ./lib { lib = devlib.lib // nixos.lib; };
+
+      sharedOverlays = [
+        (final: prev: {
+          ourlib = self.lib;
+        })
+      ];
+
+      nixos = {
+        hostDefaults = {
+          system = "x86_64-linux";
+          channelName = "nixos";
+          modules = ./modules/module-list.nix;
+          externalModules = [
+            { _module.args.ourlib = self.lib; }
+            ci-agent.nixosModules.agent-profile
+            home.nixosModules.home-manager
+            ./modules/customBuilds.nix
+          ];
+        };
+        hosts = nixos.lib.mkMerge [
+          (devlib.lib.importers.importHosts ./hosts)
+          { /* set host specific properties here */ }
```

```suggestion
          { 
            /* set host specific properties here */ 
            NixOS = { };
          }
```

I was wondering, if we should give a little (obvious) hint to users.

---

Good idea, I think we should. Just added in latest commit.